### PR TITLE
doc/user: fix cast documentation for timestamps

### LINKED ISSUE #2512: Documentation about timestamp casts is confusing
The docs for info about [valid casts to and from timestamps](https://materialize.io/docs/sql/types/timestamp/#valid-casts) seems to imply that you can convert to from timestamp to string, and later is says that you cannot cast timestamp (or timestamptz) to any other type. There is an example of casting from a string to a timestamp below, and that was the info I was looking for, but the valid-casts section would have been nice.

### PR DESCRIPTION
**Pull request by benesch**
Fix #2512.

### CODE REVIEW

**Comment by frankmcsherry** [comment]
Pretty sure you want #2512, but don't want to unilaterally change.

File: doc/user/sql/types/text.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ Detail | Info
 
 #### From `text`
 
-You cannot cast `text` to any other type.
+You can cast `text` to [all types](../), provided the string is valid input
```
Comment (frankmcsherry) [inline doc/user/sql/types/text.md:30]: You've got `[cast](../../functions/cast)` elsewhere (with a link) which seems smart! Also, if you are going to change this line anyhow, I wonder if there are words that can distinguish between which types can be cast between, and what happens when you do (I think the "provided the .." clause trips up dynamic v static typing).

Comment (benesch) [reply doc/user/sql/types/text.md:30]: > I wonder if there are words that can distinguish between which types can be cast between, and what happens when you do (I think the "provided the .." clause trips up dynamic v static typing).

I'm confused. What's incorrect about how it's worded? The cast will fail unless the string is valid input for the destination type.

**Comment by benesch** [comment]
Reworded! Feel free to submit a follow-up PR to fix the wording to your liking. I think I made it a bit clearer but I'm not sure I actually understood what was wrong with it originally!

File: doc/user/sql/types/text.md (hunk 1 above)
Comment (frankmcsherry) [reply doc/user/sql/types/text.md]: My reading of statically typed stuff is that you *can* e.g. cast from `text` to any other type, but that cast may fail if the string is not valid input. In contrast with a dynamically typed system where the resulting type may depend on whether the cast succeeds or fails.

**Comment by frankmcsherry** [comment]
The new text looks great (sorry, responded to your in-line comment before seeing it was updated).

**Pull request merged**